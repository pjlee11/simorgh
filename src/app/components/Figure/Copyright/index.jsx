import React from 'react';
import { string } from 'prop-types';
import styled from 'styled-components';
import {
  FF_NEWS_SANS_REG,
  C_WHITE,
  GEL_SPACING,
  GEL_SPACING_HLF,
} from '../../../lib/constants/styles';
import { T_MINION } from '../../../lib/constants/typography';
import VisuallyHiddenText from '../../VisuallyHiddenText';

const StyledCopyright = styled.p.attrs({
  role: 'text',
})`
  ${T_MINION};
  background-color: rgba(34, 34, 34, 0.75);
  text-transform: uppercase;
  color: ${C_WHITE};
  padding: ${GEL_SPACING_HLF} ${GEL_SPACING};
  font-family: ${FF_NEWS_SANS_REG};
  position: absolute;
  bottom: 0;
  right: 0;
  margin: 0;
`;

const Copyright = ({ children }) => {
  const attributionText = `Image source, `;

  return (
    <StyledCopyright>
      <VisuallyHiddenText>{attributionText}</VisuallyHiddenText>
      {children}
    </StyledCopyright>
  );
};

Copyright.propTypes = {
  children: string.isRequired,
};

export default Copyright;
