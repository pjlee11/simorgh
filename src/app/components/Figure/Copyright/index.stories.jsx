import React from 'react';
import { storiesOf } from '@storybook/react'; // eslint-disable-line import/no-extraneous-dependencies
import Copyright from './index';

storiesOf('Copyright', module).add('default', () => (
  <Copyright>Getty Images</Copyright>
));
