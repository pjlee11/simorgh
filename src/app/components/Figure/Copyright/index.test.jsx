import React from 'react';
import { shouldMatchSnapshot } from '../../../helpers/tests/testHelpers';
import Copyright from './index';

describe('Copyright', () => {
  shouldMatchSnapshot(
    'should render correctly',
    <Copyright>Getty Images</Copyright>,
  );
});
