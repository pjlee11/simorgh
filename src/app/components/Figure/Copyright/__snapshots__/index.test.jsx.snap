// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Copyright should render correctly 1`] = `
.c1 {
  position: absolute;
  width: 1px;
  height: 1px;
  margin: -1px;
  padding: 0;
  border: 0;
  overflow: hidden;
  -webkit-clip: rect(0 0 0 0);
  clip: rect(0 0 0 0);
  top: 0;
}

.c0 {
  font-size: 0.75em;
  line-height: 1em;
  background-color: rgba(34,34,34,0.75);
  text-transform: uppercase;
  color: #FFFFFF;
  padding: 0.25rem 0.5rem;
  font-family: ReithSansNewsRegular,Helvetica,Arial,sans-serif;
  position: absolute;
  bottom: 0;
  right: 0;
  margin: 0;
}

<p
  className="c0"
  role="text"
>
  <span
    className="c1"
  >
    Image source, 
  </span>
  Getty Images
</p>
`;
